CDD-2427: Charts-v2 documentation update

phill-stanley opened the pull request:
# Description

This PR includes updates to charts-v2 swagger documentation and error messages

Fixes #CDD-2427

---

## Type of change

Please select the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Tech debt item (this is focused solely on addressing any relevant technical debt)

---

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests at the right levels to prove my change is effective
- [ ] I have added screenshots or screen grabs where appropriate
- [ ] I have added docstrings in the correct style [(google)](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings)

A-Ashiq commented on metrics/interfaces/plots/access.py:
```diff
@@ -24,13 +24,28 @@
-class DataNotFoundForPlotError(Exception): ...
+class DataNotFoundForPlotError(Exception):
+    def __init__(self):
+        message = "No data was found for the plot requested, please review the plot parameters provided."
```
Nice

A-Ashiq commented:
<img width="737" alt="Screenshot 2025-03-03 at 10 13 21" src="https://github.com/user-attachments/assets/a3439b82-7cc7-4370-a94c-5605ebcf1eb8" />

Do you reckon its worth adding default values for these fields. Otherwise I think OpenAPI will just render in any old value. This is what confused DPD the first time around

A-Ashiq commented on metrics/api/views/charts.py:
```diff
@@ -134,6 +133,25 @@ def post(self, request, *args, **kwargs):
 
+        ---
+
+        # Choosing a y-axis minimum and maximum value (y-axis range)
```
This is really good 👍

A-Ashiq commented on metrics/interfaces/plots/access.py:
```diff
@@ -24,13 +24,28 @@
-class DataNotFoundForAnyPlotError(Exception): ...
+class DataNotFoundForAnyPlotError(Exception):
+    def __init__(self):
+        message = (
```
Nice

phill-stanley commented:
> <img alt="Screenshot 2025-03-03 at 10 13 21" width="737" src="https://private-user-images.githubusercontent.com/47219506/418528241-a3439b82-7cc7-4370-a94c-5605ebcf1eb8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDEwMTIwMTEsIm5iZiI6MTc0MTAxMTcxMSwicGF0aCI6Ii80NzIxOTUwNi80MTg1MjgyNDEtYTM0MzliODItN2NjNy00MzcwLWE5NGMtNTYwNWViY2YxZWI4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzAzVDE0MjE1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUzZGM2NDg5NTdmMTY0NDdmNGY4MmQ4NzExZjJiOGI1NTJmNTZlODliN2ZjODE4NGEyZGI0YjAzM2QzOGRhOWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.oxnv_YN5CG__O97TkkALRSqL6RyWGpCZyOa9L1goflA">
> Do you reckon its worth adding default values for these fields. Otherwise I think OpenAPI will just render in any old value. This is what confused DPD the first time around

Yeah good shout, I've updated this and added in a default example.
